lib/redis/cluster/command_loader.rb

```diff
@@ -10,15 +10,15 @@ module CommandLoader
       module_function
 
       def load(nodes)
-        nodes.each do |node|
+        errors = nodes.map do |node|
           begin
             return fetch_command_details(node)
-          rescue CannotConnectError, ConnectionError, CommandError
-            next # can retry on another node
+          rescue CannotConnectError, ConnectionError, CommandError => e
+            next e
```

Comment:
nitpick, I think that `next` is making the code confusing.

Since it's on the last line you can  just let the expression return by itself.

```suggestion
          rescue CannotConnectError, ConnectionError, CommandError => error
            error
```